src/dss/create_notebook.py

```diff
@@ -0,0 +1,74 @@
+from pathlib import Path
+
+from charmed_kubeflow_chisme.kubernetes import KubernetesResourceHandler
+from lightkube import Client
+from lightkube.resources.apps_v1 import Deployment
+from lightkube.resources.core_v1 import Service
+
+from dss.config import (
+    DSS_CLI_MANAGER_LABELS,
+    DSS_NAMESPACE,
+    FIELD_MANAGER,
+    MANIFEST_TEMPLATES_LOCATION,
+    NOTEBOOK_PVC_NAME,
+)
+from dss.logger import setup_logger
+from dss.utils import get_mlflow_tracking_uri, get_notebook_url, wait_for_deployment_ready
+
+# Set up logger
+logger = setup_logger("logs/dss.log")
+
+
+def create_notebook(name: str, image: str, lightkube_client: Client) -> None:
+    """
+    Creates a Notebook server on the Kubernetes cluster.
+
+    Args:
+        name (str): The name of the notebook server.
+        image (str): The image used for the notebook server.
+        lightkube_client (Client): The Kubernetes client.
+
+    Returns:
+        None
+    """
+    manifests_file = Path(
+        Path(__file__).parent, MANIFEST_TEMPLATES_LOCATION, "notebook_deployment.yaml.j2"
+    )
```

Comment:
to make sure I understand, do you mean in `config.py` we have things like
```python
NOTEBOOK_DEPLOYMENT_MANIFEST = Path(Path(__file__) ... "notebook_deployment.yaml.j2")
MLFLOW_DEPLOYMENT_MANIFEST = Path(Path(__file__) ... "mlflow_deployment.yaml.j2")

```
?

I'm cool with that.  My only worry is how `__file__` will work in that case.  Would need to test

Reply:
Yeah, I meant exactly that. We would need to change slightly the path, but I think we can sort it out.

Reply:
I tested to make sure this would work and I think it does.  If we write in `config.py`:
```python
MANIFEST_TEMPLATES_LOCATION = Path("./manifest_templates")
ABSOLUTE_MANIFEST_TEMPLATES_LOCATION = Path(__file__).parent / MANIFEST_TEMPLATES_LOCATION
```

then the `__file__` in that call is relative to `config.py`, not `create_notebook.py` or whoever is importing `config.py`.  But I think that's ok.

Where this is following the same pattern as the other commands, do you mind if we implement this change separate to this PR and do it for all commands at once?  We'd have the same change for `initialize.py` and probably what Noha is implementing.

Reply:
i filed issue https://github.com/canonical/data-science-stack/issues/61 to track this
@DnPlas resolve the conversation if you're okay with that